src/main/java/github/scarsz/discordsrv/DiscordSRV.java

```diff
@@ -1768,9 +1765,13 @@ public void broadcastMessageToMinecraftServer(String channel, Component message,
                 .map(hook -> (ChatHook) hook)
                 .findAny().orElse(null);
 
+        DiscordGuildMessagePreBroadcastEvent preBroadcastEvent = api.callEvent(new DiscordGuildMessagePreBroadcastEvent
+                (channel, message, PlayerUtil.getOnlinePlayers()));
+        message = preBroadcastEvent.getMessage();
+        channel = preBroadcastEvent.getChannel();
         if (chatHook == null || channel == null) {
             if (channel != null && !channel.equalsIgnoreCase("global")) return; // don't send messages for non-global channels with no plugin hooks
-            MessageUtil.sendMessage(PlayerUtil.getOnlinePlayers(), message);
+            MessageUtil.sendMessage(preBroadcastEvent.getRecipients(), message);
             PlayerUtil.notifyPlayersOfMentions(null, MessageUtil.toLegacy(message));
         } else {
```

Comment:
You should either not run the event when chat hooks are in use, or add documentation/throw a exception for trying to change recipients when chat hooks are being used.

Reply:
I hadn't thought of that. I'll move it so it doesn't affect any chat hooks.